src/transport/SessionManager.cpp

```diff
@@ -201,6 +201,17 @@ CHIP_ERROR SessionManager::PrepareMessage(const SessionHandle & sessionHandle, P
         packetHeader.SetSecureSessionControlMsg(true);
     }
 
+    if (sessionHandle->AllowsMRP())
+    {
+        VerifyOrReturnError(message->TotalLength() <= kMaxAppMessageLen, CHIP_ERROR_MESSAGE_TOO_LONG);
+    }
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    else if (sessionHandle->AllowsLargePayload())
```

Comment:
So a few questions here:

1. Should we not enforce `<= kMaxAppMessageLen` for _all_ transports that don't allow large payloads, whether they allow MRP or not?
2. Why is this conditioned on `INET_CONFIG_ENABLE_TCP_ENDPOINT`?   Is that just an attempt to save some codesize when TCP is not supported?  If so, I would prefer that were handled by having `AllowsLargePayload` compile-time conditional or something (done in a followup), not by sprinkling ifdefs on its consumers.